molo/core/models.py

```diff
@@ -556,13 +556,57 @@ class LanguageRelation(models.Model):
     language = models.ForeignKey('core.SiteLanguage', related_name='+')
 
 
+def get_translation_for_new_qs(qs, locale, site, is_live=True):
+    translated_pages = []
+    language = Languages.for_site(site).languages.get(locale=locale)
+    for article in qs:
+        cache_key = article.get_translation_for_cache_key(
+            locale, site, is_live)
+        trans_pk = cache.get(cache_key)
+
+        # TODO: consider pickling page object. Be careful about page size in
+        # memory
+        if trans_pk:
+            translated_pages.append(Page.objects.get(pk=trans_pk).specific)
+            continue
+        try:
+            translated_page = article.translated_pages.get(language=language)
+            cache.set(cache_key, translated_page.pk, None)
+            translated_pages.append(translated_page)
+        except:
+            translated_pages.append(article)
+    return translated_pages
+
+
 class TranslatablePageMixinNotRoutable(object):
     def get_translation_for_cache_key(self, locale, site, is_live):
         return "get_translation_for_{}_{}_{}_{}_{}".format(
             self.pk, locale, site.pk, is_live,
             self.latest_revision_created_at.isoformat())
 
     def get_translation_for(self, locale, site, is_live=True):
+        if settings.USE_NEW_TRANSLATIONS:
+            return self.get_translation_for_new(locale, site, is_live)
+        return self.get_translation_for_old(locale, site, is_live)
+
+    def get_translation_for_new(self, locale, site, is_live=True):
```

Comment:
should the default for is_live not be False?

Reply:
No. by default we only want to get the live pages.

Reply:
I thought is live meant like whether you are in production

Reply:
nope. it's a flag to show if we only want the translated pages that are live, ie they have been published